### PR TITLE
feat(analytics): walk-up-to-clickable-parent for autocapture

### PR DESCRIPTION
**Pull request by rahul-mixpanel**
## Summary
- When a non-interactive leaf view is tapped, walk up the view hierarchy to the nearest clickable ancestor and use its identity for `$el_id`
- Walk-up is always-on and uses interactivity checks (`isClickable()`/`isLongClickable()`), not identity checks — consistent with iOS behavior
- Extract `resolveIdentity()` from `resolveElementId()` to eliminate duplicate resource lookups per tap
- Fix ancestor hierarchy string to be built from the ancestor view, not the leaf
- Add walk-up demo test screens for both XML and Compose
- Add instrumented tests validating walk-up behavior across XML and Compose views

## Test plan
- [x] Walk-up resolves to clickable parent's `$el_id` for non-interactive leaves
- [x] Non-interactive leaf with own `contentDescription` still walks up (leaf identity ignored)
- [x] Clickable leaf does NOT walk up (owns its own click)
- [x] Nested clickables stop at the innermost clickable parent
- [x] Deep nesting (9 levels) finds clickable ancestor within 10-level limit
- [x] Non-interactive leaf with no clickable ancestor falls back to hash
- [x] Android build compiles successfully
- [ ] Run `./gradlew :analytics:connectedAndroidTest` on device/emulator

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: analytics/src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java [hunk 1]
```diff
@@ -1083,104 +1083,6 @@ private void initializeAutocapture(AutocaptureOptions options) {
     }
 
      /**
-     * Provides methods for tracking autocapture events. Events tracked through this class
-     * are automatically tagged with the {@code $mp_autocapture} property, which causes them
-     * to appear with an "[Auto]" prefix in the Mixpanel web app.
-     *
-     * <p>Access this via {@link MixpanelAPI#getAutocapture()}.
-     */
-    public class Autocapture {
```
Comment (rahul-mixpanel) [inline analytics/src/main/java/com/mixpanel/android/mpmetrics/MixpanelAPI.java:1092]: The previous PR ended up with duplicate `Autocapture` classes due to a rebase.

File: analytics/src/main/java/com/mixpanel/android/autocapture/SemanticExtractor.java [hunk 2]
```diff
@@ -0,0 +1,802 @@
...
+
+        // Walk up to nearest clickable ancestor
+        ViewParent parent = tappedView.getParent();
+        int depth = 0;
+        while (parent instanceof View && depth < AutocaptureDefaults.MAX_ANCESTOR_SEARCH_DEPTH) {
+            View ancestor = (View) parent;
+            if (ancestor.isClickable() || ancestor.isLongClickable()) {
+                // Found a clickable ancestor — rebuild hierarchy from its perspective
+                return extractFromView(ancestor, buildHierarchyString(ancestor), x, y);
+            }
```
[the remark this answers is not in the document]
Comment (tylerjroach) [reply analytics/src/main/java/com/mixpanel/android/autocapture/SemanticExtractor.java]: This is valid.

File: analytics/src/main/java/com/mixpanel/android/autocapture/SemanticExtractor.java [hunk 3]
```diff
@@ -739,6 +759,43 @@ private static String getSimpleClassName(@NonNull String fullyQualifiedName) {
...
+     * not identity. A non-interactive leaf (e.g., TextView inside a clickable
+     * LinearLayout) will always resolve to its clickable parent's identity,
+     * even if the leaf has its own contentDescription.
+     */
+    @NonNull
+    private static ClickEvent.Builder walkUpToClickableParent(
+            @NonNull View tappedView, @NonNull ClickEvent.Builder original,
+            @Nullable String hierarchy, float x, float y) {
+        // Only walk up if the tapped view is not interactive
+        if (tappedView.isClickable() || tappedView.isLongClickable()) {
```
Comment (tylerjroach) [inline analytics/src/main/java/com/mixpanel/android/autocapture/SemanticExtractor.java]: Can we add isContextClickable as well. Edge case but it is another option. Can you also add a test for a CheckBox and RadioButton, etc. `isCheckable` is available on these, and I'm not sure if they report as clickable or not. If not, we may need to type check for these.

Should point out that if a customer made their own touchListener, we may walk up when we shouldn't, but I think this is more of an edge case, and we have to make best guesses here.

**Review by tylerjroach — CHANGES_REQUESTED** [review]
small addition ask.

File: analytics/src/main/java/com/mixpanel/android/autocapture/SemanticExtractor.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (rahul-mixpanel) [reply analytics/src/main/java/com/mixpanel/android/autocapture/SemanticExtractor.java]: When a clickable card (for example, a cart item) contains a disabled **Delete** button, tapping the disabled button should trigger a dead click and capture the button’s identity, not the parent card’s. That’s why we intentionally skip the enabled check here.

File: analytics/src/main/java/com/mixpanel/android/autocapture/SemanticExtractor.java (hunk 3 above)
Comment (rahul-mixpanel) [reply analytics/src/main/java/com/mixpanel/android/autocapture/SemanticExtractor.java]: Good call raising `isContextClickable` — I looked into it and decided to leave it out intentionally. Context clicks come from right-click or stylus secondary button interactions, which are separate from taps. Since this walk-up only runs on the tap capture path, a view that's context-clickable but not tap-clickable wouldn't produce a tap event anyway, so including it wouldn't change behavior in practice. This holds in the disabled case too — `isClickable()` still returns `true` on a disabled view, so walk-up correctly stops there regardless of context-clickability.

For `isCheckable`, added `instanceof Checkable` to both guards in `walkUpToClickableParent` to match the accessibility node path. Also added tests for `CheckBox` and `RadioButton` — they're both clickable by default so they correctly keep their own identity without walking up.

**Review by tylerjroach — APPROVED** [review]
